includes/class-freemius.php

```diff
@@ -253,19 +266,33 @@ class Freemius extends Freemius_Abstract {
 		 * @author Vova Feldman (@svovaf)
 		 * @since  1.0.0
 		 *
+		 * @param number $module_id
 		 * @param string $slug
-		 * @param bool   $is_init Since 1.2.1 Is initiation sequence.
+		 * @param bool   $is_init Since 1.2.2 Is initiation sequence.
 		 */
-		private function __construct( $slug, $is_init = false ) {
-			$this->_slug = $slug;
+		private function __construct( $module_id, $slug = false, $is_init = false ) {
+			if ( ! is_numeric( $module_id ) ) {
+				$slug = $module_id;
 
-			$this->_logger = FS_Logger::get_logger( WP_FS__SLUG . '_' . $slug, WP_FS__DEBUG_SDK, WP_FS__ECHO_DEBUG_SDK );
+				$this->_plugin = FS_Plugin_Manager::instance( $slug, $slug, WP_FS__MODULE_TYPE_PLUGIN )->get();
```

Comment:
@fajardoleo I'm not sure why you שרק loading the plugin's data here if you already loading it later in the constructor
`$this->_plugin = FS_Plugin_Manager::instance( $this->_module_id, $this->_slug, $this->_module_type )->get();`